File: jan-11-25/sub.py
from collections import Counter

class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if len(s) < k:
            return False
        if len(s) == k:
            return True

        odd = sum([cnt & 1 for k,cnt in Counter(s).items()])

        return odd <= k

        

File: jan-11-25/sub.java
class Solution {
    public boolean canConstruct(String s, int k) {
        if(s.length() < k) return false;
        if(s.length() == k) return true;

        int mp[] = new int[26];

        for(char i: s.toCharArray()){
            mp[i - 'a'] ^= 1;
        }

        for(int i: mp) k -= i;

        return k > -1;
    }
}